src/dotnet/UI.Blazor.App/Components/ChatMessageEditor/NotifyAllButton.razor

```diff
@@ -0,0 +1,144 @@
+@using ActualChat.Notification
+@namespace ActualChat.UI.Blazor.App.Components
+@inherits ComputedStateComponent<NotifyAllButton.Model>
+@{
+    var m = State.Value;
+    if (!m.IsVisible)
+        return;
+}
+
+<ButtonRound
+    Class="post-panel-btn notify-call-btn btn-xs btn-transparent btn-primary"
+    Tooltip="Notify All"
+    TooltipPosition="FloatingPosition.Top"
+    Click="@OnNotifyAllClick">
+    <i class="icon-notify-phone text-2xl"></i>
+</ButtonRound>
+
+@code {
+    private (ChatEntry, Moment)? _lastKnownOwnUnreadEntryInfo;
+
+    [Inject] private ChatUIHub Hub { get; init; } = null!;
+    private Session Session => Hub.Session();
+    private AccountUI AccountUI => Hub.AccountUI;
+    private IAuthors Authors => Hub.Authors;
+    private IChats Chats => Hub.Chats;
+    private ModalUI ModalUI => Hub.ModalUI;
+    private UICommander UICommander => Hub.UICommander();
+    private MomentClock Clock => Hub.Clocks().SystemClock;
+    private LiveTime LiveTime => Hub.LiveTime;
+
+    [Parameter, EditorRequired] public Chat Chat { get; set; } = null!;
+
+    protected override ComputedState<Model>.Options GetStateOptions()
+        => new() {
+            InitialValue = new Model(Chat.Id.Kind is ChatKind.Peer),
+            Category = ComputedStateComponent.GetStateCategory(GetType()),
+        };
+
+    protected override async Task<Model> ComputeState(CancellationToken cancellationToken) {
+        var chatId = Chat.Id;
+        if (chatId.Kind is ChatKind.Peer)
+            return new Model(true); // Always allowed for peer chats.
+
+        var ownAuthor = Chat.Rules.Author;
+        if (ownAuthor is null)
+            return new Model(false); // Disabled if no own author.
+
+        var chatIdRange = await Chats
+            .GetIdRange(Session, chatId, ChatEntryKind.Text, cancellationToken)
+            .ConfigureAwait(false);
+        if (chatIdRange.IsEmpty)
+            return new Model(false); // Disabled if no messages yet.
+
+        var authorIds = await Authors.ListAuthorIds(Session, chatId, cancellationToken).ConfigureAwait(false);
```

Comment:
afair we agreed to extract all properties and fields into local variables boefore using passing them to async methods

Reply:
Yep, all except Session. I have to store Chat to local variable.